### DISCUSSION TITLE
How do I fix the .extern section if I messed it up by accident?

### DISCUSSION THREAD

**Discussion by VisualEhrmanntraut** [Vector35/binaryninja-api#5848] [Q&A] [answered]
I defined some data variable that was too big using the console and then went back and undid it, after saving the database I realised that the .extern section got messed up and it's only showing two functions now. How do I fix it?

**Comment by plafosse** [comment]
Its kind of hard to know what to do a this point you might have to just re-open the binary. If you could share screen shots or binaries we might be able to help more, but doing unsupported actions via the API and messing up your an analysis is kind of can't be helped.

**Reply by VisualEhrmanntraut** [reply]
I didn't do something unsupported, it just crossed the section and wrote over .extern and after undoing from the history tab the section did not fix itself. I found I can redefine an external symbol if I can access it from the view somehow like from a call in the function.

**Reply by plafosse** [reply, marked as answer]
Sorry this got buried. Could you provide a set of repro steps?

**Reply by psifertex** [reply]
Without repro steps it's hard to answer so marking this as closed and answered for now. Feel free to re-open with repro steps.
